flow/connectors/utils/aws.go

```diff
@@ -178,6 +201,20 @@ func GetAWSCredentialsProvider(ctx context.Context, connectorName string, peerCr
 	return NewConfigBasedAWSCredentialsProvider(awsConfig), nil
 }
 
+func FileURLForS3Service(endpoint string, region string, bucket string, filePath string) string {
+	fileUrl := fmt.Sprintf("https://%s.s3.%s.amazonaws.com/%s", bucket, region, filePath)
+	if strings.Contains(endpoint, "storage.googleapis.com") {
```

Comment:
Don't like this, ideally the regex would below would handle if user enters `https://storage.googleapis.com` & we're pretty much ignoring the endpoint when it contains this specific string